### PR TITLE
w 7pm b replace existing classes & schedule remove button

### PR DESCRIPTION
**Pull request by Zonkman**
the "add" method of CourseController has been changed to replace the old lecture and section, if it finds an old one. 

also, because of moving common delete functionality to another function, and it having already existed in the delete method, this absorbed <https://github.com/ucsb-cs56-w20/ucsb-courses-search/pull/237> . so this request also made the delete button on the schedule page for each class, that when pressed, removes the lecture and section together from the schedule.

### CODE REVIEW

**Comment by Zonkman** [comment]
merge conflicts fixed, my methods now use the new names, and functionality combined

File: src/main/java/edu/ucsb/cs56/ucsb_courses_search/controller/CourseController.java [hunk 1]
```diff
@@ -64,6 +65,22 @@ public String index(Model model, OAuth2AuthenticationToken token) throws AccessF
         }
         return "courseschedule/index";
     }
+
+    /**
+     * Deletes all courses with the same name as "name" from "courseList" and CourseRepository.
+     * Returns a list without any courses of name "name".
+    */
+    private List<ScheduleItem> deleteByClassname(List<ScheduleItem> courseList, String name){
```
Comment (agprice) [inline src/main/java/edu/ucsb/cs56/ucsb_courses_search/controller/CourseController.java]: What is this function doing in the controller? Controllers are only for mappings. If you want to do something like this, it should probably be bundled into a service. Also, I'm not sure I see the reason for duplicating the list, when you could just modify the courseList passed in to the function.

File: src/main/java/edu/ucsb/cs56/ucsb_courses_search/controller/CourseController.java [hunk 2]
```diff
@@ -90,9 +106,28 @@ public String add(ScheduleItem scheduleItem,
...
-        model.addAttribute("myclasses", scheduleItemRepository.findByUid(scheduleItem.getUid()));
+        model.addAttribute("myclasses", myClasses);
+        return "courseschedule/index";
+    }
+
+    @PostMapping("/courseschedule/delete")
+    public String delete(ScheduleItem scheduleItem, Model model){
+        logger.info("ScheduleItem to delete: " + scheduleItem.getUid());
+        List<ScheduleItem> myClasses = scheduleItemRepository.findByUid(scheduleItem.getUid());
+        List<ScheduleItem> remainingClasses = deleteByClassname(myClasses, scheduleItem.getClassname());
```
Comment (agprice) [inline src/main/java/edu/ucsb/cs56/ucsb_courses_search/controller/CourseController.java]: If deleteByClassname simply modified the list passed to it, then you wouldn't need two separate lists. Is there a reason this approach wasn't taken?

File: src/main/java/edu/ucsb/cs56/ucsb_courses_search/controller/CourseController.java (hunk 1 above)
Comment (Zonkman) [reply src/main/java/edu/ucsb/cs56/ucsb_courses_search/controller/CourseController.java]: had trouble adding the service (still looking into it), also i just forgot about passing by reference when i wrote that

**Comment by agprice** [comment]
When trying to add a class, a whitelabel error page is produced.

**Pull request merged**